source/upgrade.txt

```diff
@@ -0,0 +1,101 @@
+.. _cpp-upgrade:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: compatibility, backwards compatibility
+
+Overview
+--------
+
+This page describes the changes you must make to your application
+when you upgrade to a new version of the {+driver-short+}.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the {+mdb-server+} versions
+  your application connects to and the C++ version your
+  application runs on. For version compatibility information, see the
+  :ref:`{+driver-short+} Compatibility <cpp-compatibility-tables>`
+  page.
+- Address any breaking changes between the driver version
+  your application is using and your planned upgrade version as detailed in the
+  :ref:`Breaking Changes <cpp-breaking-changes>` section.
+
+
+.. tip::
+
+   To minimize the number of changes your application requires when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <cpp-stable-api>`.
+
+.. _cpp-breaking-changes:
+
+Breaking Changes
+----------------
+
+.. note::
+
+   The {+driver-short+} doesn't ensure API or ABI stability across patch releases
+   within a minor version.
+
+A breaking change is a change of a convention or a behavior starting in a specific
+version of the driver. This type of change may prevent your application from working
+properly if not addressed before upgrading the driver.
+
+The breaking changes in this section are categorized by the driver version that introduced
+them. When upgrading driver versions, address all the breaking changes between the current
+and upgrade versions.
+
+.. example:: Upgrading from Version 3.0
+
+  If you're upgrading the {+driver-short+} from version 3.0 to version 3.10,
+  address all breaking changes listed for versions 3.1 to 3.10, if any.
+
+.. _version-3.10-breaking-changes:
```

Comment:
> This page includes breaking changes only for versions >= 3.9, since that's when the team began maintaining a changelog (and v3.0, which is flatly incompatible with prior versions).

Suggest removing mentions of any versions prior to 3.9 if the intent is to avoid having to translate [prior release notes](https://github.com/mongodb/mongo-cxx-driver/releases#hd-a8ede57b) into a log of potential build-breaking changes. This jump from 3.0 to 3.9 may only confuse users due to the absence of any intermediate releases.

Reply:
Debated doing this initially; glad to hear you agree v3.0 information would be more confusing than helpful.